.github/workflows/ci.yml

```diff
@@ -33,6 +33,9 @@ env:
 permissions:
   statuses: write
   id-token: write # This is required for requesting the JWT/OIDC
+  # clippy bot requires write permission for contents and pull-requests
+  contents: write
```

Comment:
is there some documentation you can link to on what these permissions mean? Just want to make sure we aren't giving it permission to modify source code, etc

Reply:
I hypothesize that the reason why I never ran into this problem is [because I never added a permissions parameter to my clippybot.yml file](https://github.com/maddeleine/s2n-quic/blob/main/.github/workflows/clippybot.yml). 
The ci.yml has a permissions parameter, maybe once you add it, it removes scope to only what is listed.